packages/colibri/src/project_manager/tool/quartus/quartusProjectManager.ts

```diff
@@ -0,0 +1,90 @@
+// This code only can be used for Quartus boards
+
+import { t_action_result, t_file } from "../../common";
+import { Project_manager } from "../../project_manager";
+import { get_toplevel_from_path } from "../../../utils/hdl_utils"
+import * as chokidar from "chokidar";
+import {
+    QuartusExecutionError, addFilesToProject, removeFilesFromProject, setTopLevelPath,
+    getProjectInfo, getFilesFromProject
+} from "./utils";
+
+export class QuartusProjectManager extends Project_manager {
+
+    private projectPath: string;
+    private quartusProjectWatcher: chokidar.FSWatcher;
+
+    constructor(name: string, projectPath: string) {
+        super(name);
+        this.projectPath = projectPath;
+
+        this.quartusProjectWatcher = chokidar.watch('file', {
+            usePolling: true, interval: 2000
+        });
+        this.quartusProjectWatcher.on('change', (_path, _stats) => {
+            this.refreshFromQuartusProject();
+        });
+    }
+
+    async refreshFromQuartusProject() {
```

Comment:
it shouldn't be any specific function, just functions in Project_manager overwritten here

Reply:
yes, I need to think how to do it in a next PR

Reply:
you are agree